### PR TITLE
[typescript] use @babel/plugin-proposal-object-rest-spread

### PR DESCRIPTION
**Pull request by resir014**
Recommended by TypeScript Babel Starter:
https://github.com/Microsoft/TypeScript-Babel-Starter

### CODE REVIEW

File: packages/gatsby-plugin-typescript/package.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
     "Noah Lange <noahrlange@gmail.com>"
   ],
   "dependencies": {
+    "@babel/plugin-proposal-object-rest-spread": "^7.0.0",
     "@babel/preset-typescript": "^7.0.0",
     "@babel/runtime": "^7.0.0",
     "babel-plugin-remove-graphql-queries": "^2.0.2-rc.3"
```
Comment (resir014) [inline packages/gatsby-plugin-typescript/package.json]: This plugin doesn't seem to use this package anymore. Is it safe to remove?

Comment (pieh) [reply packages/gatsby-plugin-typescript/package.json]: Should be safe - this plugins should be added by gatsby core : https://github.com/gatsbyjs/gatsby/blob/dfeea37ac3e64b40752fb399d8c9d161ded71169/packages/gatsby/src/utils/babel-loader-helpers.js#L26-L31

**Comment by pieh** [comment]
Discussion starter - not a definitive "no":
I personally am against adding this here as "object-rest-spread" has nothing to do with typescript itself and it might cause more confusion why rest spread works with typescript plugin and doesn't work without it (or with flow plugin). People are able to add their custom babel config either via gatsby-node api or with .babelrc files ( https://next.gatsbyjs.org/docs/babel/ ). Am I missing something here? Does my argument have sense? Please discuss

**Comment by resir014** [comment]
@pieh I remembered not being able to use object spread back in the early days of TS 2.x (i.e. it will straight out give a compile error). The TypeScript team has a hard rule of not implementing new ES features until it reaches stage-3, so newer stuff like optional chaining is not supported through their compiler/type checker.

Now that rest has reached [stage-4](https://github.com/tc39/proposal-object-rest-spread), in later versions of TS you can actually use them, so it should work well without these proposal plugins, and I understand if we shouldn't pollute the build step with unneeded Babel presets.

**Comment by pieh** [comment]
I'm not saying that it will or will not work (don't have enough context) - it's just I don't feel like `gatsby-plugin-typescript` is right place for adding babel plugin that is unrelated to typescript itself

**Comment by resir014** [comment]
In which case - you're probably right. Will need more people's opinions on this.

**Comment by KyleAMathews** [comment]
Yeah, let's not do this. It'd be unexpected for the Typescript plugin to add additional babel plugins.

Doesn't our default Babel setup include this plugin though anyways?

**Pull request closed without merging**